test/features/interceptor/authorization_interceptor_test.dart

```diff
@@ -0,0 +1,362 @@
+          OIDCFixtures.oidcConfiguration.scopes,
+          OIDCFixtures.tokenOidcExpired.refreshToken
+        )).thenAnswer((_) async => OIDCFixtures.tokenOidc);
+
+        when(accountCacheManager.getCurrentAccount()).thenAnswer((_) async => AccountFixtures.aliceAccount);
+
+        dioAdapter
+          ..onPost(
+            baseUrl,
+            (request) => request.throws(401, dioError401),
+          )
+          ..onPost(
+            baseUrl,
+            (request) => request.reply(200, {'message': 'Retry upload attachment request successfully!'})
+          );
+
+        final response = await dio.post(baseUrl);
+
+        expect(response.data, {'message': 'Retry upload attachment request successfully!'});
+      });
+
+      test('The request should not be blocked when call 2 post request with response error 401 - 401', () async {
+        final dioError401 = DioError(
+          error: {'message': 'Token Expired'},
+          requestOptions: RequestOptions(path: baseUrl),
+          response: Response(
+            statusCode: 401,
+            requestOptions: RequestOptions(path: baseUrl),
+          ),
+          type: DioErrorType.badResponse,
+        );
+
+        final dioErrorBadCertificate = DioError(
+          error: {'message': 'Bad Certificate'},
+          requestOptions: RequestOptions(path: baseUrl),
+          response: Response(
+            statusCode: 401,
+            requestOptions: RequestOptions(path: baseUrl),
+          ),
+          type: DioErrorType.badCertificate,
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        when(authenticationClient.refreshingTokensOIDC(
+          OIDCFixtures.oidcConfiguration.clientId,
+          OIDCFixtures.oidcConfiguration.redirectUrl,
+          OIDCFixtures.oidcConfiguration.discoveryUrl,
+          OIDCFixtures.oidcConfiguration.scopes,
+          OIDCFixtures.tokenOidcExpired.refreshToken
+        )).thenAnswer((_) async => OIDCFixtures.tokenOidcWithRefreshTokenEmpty);
+
+        when(accountCacheManager.getCurrentAccount()).thenAnswer((_) async => AccountFixtures.aliceAccount);
+
+        dioAdapter.onPost(
+          baseUrl,
+          (request) => request.throws(401, dioError401),
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcWithRefreshTokenEmpty,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        dioAdapter.onPost(
+          baseUrl,
+          (request) => request.throws(401, dioErrorBadCertificate),
+        );
+
+        expect(
+          () async => await dio.post(baseUrl),
+          throwsA(isA<DioError>()),
+        );
+
+        expect(
+          () async => await dio.post(baseUrl),
+          throwsA(predicate<DioError>((error) => error.type == DioErrorType.badCertificate)),
+        );
```

Comment:
IMO no need for this, because we also get refresh token inside onErrir

Reply:
No, we must determine exactly whether the returned error meets expectations or not.